frontend/src/components/MapView/MapTooltip/index.tsx

```diff
@@ -6,15 +6,86 @@ import {
   withStyles,
   WithStyles,
   LinearProgress,
+  Typography,
 } from '@material-ui/core';
-import { isEqual } from 'lodash';
-import { tooltipSelector } from 'context/tooltipStateSlice';
+import { isEmpty, isEqual, sum } from 'lodash';
+import { PopupData, tooltipSelector } from 'context/tooltipStateSlice';
 import { isEnglishLanguageSelected, useSafeTranslation } from 'i18n';
+import { TFunction } from 'utils/data-utils';
+import { ClassNameMap } from '@material-ui/styles';
+
+// This function prepares phasePopulationTable for rendering and is specific
+// to the data structure of the phase classification layer.
+// Note - this is a bit hacky for now and will likely need to be revamped if we
+// encounter other complex needs for tooltips.
+const generatePhasePopulationTable = (
```

Comment:
We could add `JSX.Element` as return value for this method ? We could also make a separate component file but it depends on the revamp you have in mind